src/components/ChangePasswordForm/ChangePasswordForm.tsx

```diff
@@ -0,0 +1,172 @@
+import { Form, Formik } from "formik";
+import React, { useState } from "react";
+import * as yup from "yup";
+import { useAuth } from "../../hooks/useAuth";
+import { PrimaryButton } from "../buttons/PrimaryButton/PrimaryButton";
+import { Input } from "../Input/Input";
+import { Spinner } from "../Spinner/Spinner";
+import {
+  ChangePasswordRequest,
+  ChangePasswordResponse,
+  JsonResponseStatus,
+} from "../../types";
+import { config } from "../../config/config";
+import classes from "./ChangePasswordForm.module.css";
+
+export interface InitialValues {
+  password: string;
+  newPassword: string;
+  confirmNewPassword: string;
+}
+
+const initialValues: InitialValues = {
+  password: "",
+  newPassword: "",
+  confirmNewPassword: "",
+};
+
+const ValidationSchema = yup.object().shape({
+  password: yup.string().required("Podaj aktualne hasło!"),
+  newPassword: yup.string().required("Podaj nowe hasło!"),
+  confirmNewPassword: yup
+    .string()
+    .required("Powtórz nowe hasło!")
+    .oneOf([yup.ref("newPassword")], "Powtórz poprawnie nowe hasło!")
+    .notOneOf([yup.ref("password")], "Nowe hasło musi być inne!"),
+});
+
+const fetchChangePassword = async (
+  data: ChangePasswordRequest
+): Promise<ChangePasswordResponse> => {
+  const response = await fetch(config.apiUrl + "auth/changePassword", {
+    method: "POST",
+    credentials: "include",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data),
+  });
```

Comment:
Wydaje mi się, że często jest podobny kod, no bo tak samo się obsługuje API. W związku z tym można to wydzielić do jednej funkcji.